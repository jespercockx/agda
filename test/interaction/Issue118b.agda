

data A : Set where
  a : A

data B : A → Set where
  b : B a

mutual

  a′ : A
  a′ = {!f b!}

  f : B a′ → A
  f b = a
