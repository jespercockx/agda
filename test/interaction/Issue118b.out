(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals, Errors*" "?0 : A ———— Errors ———————————————————————————————————————————————— Failed to solve the following constraints: Check definition of f : B a′ → A stuck because Issue118b.agda:15,5-6 I'm not sure if there should be a case for the constructor b, because I get stuck when trying to solve the following unification problems (inferred index ≟ expected index): a ≟ a′ when checking that the pattern b has type B a′ (blocked on _8) " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "1,3-4 a != a′ of type A when checking that the expression b has type B a′" nil)
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
